roachtest: update pg_regress

yuzefovich opened the pull request:
Fixes: #121405.

yuzefovich committed:
roachtest: remove custom patch for setseed in pg_regress

We do support `setseed` builtin now. However, I'd guess that values
produced by `random()` after `setseed()` for us would still be different
than in postgres (i.e. this change will likely increase the pg_regress
diff once updated).

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/121524)
<!-- Reviewable:end -->

yuzefovich committed:
roachtest: update pg_regress diff

This also includes the DEFAULT parameter work that is not yet merged
(but should be very soon).

Release note: None

michae2 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Thank you!

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121524#-:-Nu_aYIV6XzHBfay5OJ4:b-ae66e1)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rytaft)



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
